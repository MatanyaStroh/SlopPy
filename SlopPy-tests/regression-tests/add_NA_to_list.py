# try to add an NA object to a list in various ways;
# all attempts should get rejected

x = 1 / 0
assert type(x) is NA

l = []
l.append(x)
assert len(l) == 0
l.insert(0, x)
assert len(l) == 0
l.extend([x])
assert len(l) == 0
l += [x]
assert len(l) == 0

l2 = ['dummy', 'dummy2']
l2[0] = x
assert l2 == ['dummy', 'dummy2']
l2[0:1] = x
assert l2 == ['dummy', 'dummy2']

# list constructor should ignore all instances of x
assert [x] == []
assert [x,x] == []
assert [x,x,x] == []
assert [1,2,x,3,4] == [1,2,3,4]
assert [1,2,x,3,4,x] == [1,2,3,4]
assert [x,1,2,x,3,4,x] == [1,2,3,4]
assert [x,x,x,x,x,1,x] == [1]

