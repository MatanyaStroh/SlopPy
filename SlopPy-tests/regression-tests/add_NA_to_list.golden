{NA <type 'exceptions.ZeroDivisionError'> 'integer division or modulo by zero'
 Traceback (most recent call last):
  File "regression-tests/add_NA_to_list.py", line 4, in <module>
    x = 1 / 0
}
{list.append(NA) | regression-tests/add_NA_to_list.py <module>:8}
{list.insert(NA) | regression-tests/add_NA_to_list.py <module>:10}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_list.py <module>:12}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_list.py <module>:14}
{PyObject_SetItem(value=NA) | regression-tests/add_NA_to_list.py <module>:18}
{assign_slice(NA) | regression-tests/add_NA_to_list.py <module>:20}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_list.py <module>:24}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_list.py <module>:25}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_list.py <module>:26}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_list.py <module>:27}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_list.py <module>:28}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_list.py <module>:29}
{BUILD_LIST(* NA *) | regression-tests/add_NA_to_list.py <module>:30}
